### ISSUE TITLE
End of support for pre-built binaries for some os/arch

### ISSUE THREAD

**Issue by wata727** [terraform-linters/tflint#1219]
## TL;DR;

End of support for pre-built binaries for the following os/arch:

- All arch for FreeBSD/OpenBSD/NetBSD
- windows/arm and windows/arm64
- darwin/386

## Details

TFLint currently supports 20+ pre-built binaries. However, this was not planned and was originally set to follow the GoReleaser defaults. As a result, there are some issues:

- Increased release build time
  - `release/goreleaser` action takes more than 20min
- Unclear os/arch support
  - Since the supported binaries are not clear, it has been increased or decreased due to the version upgrade of Go/GoReleaser, and it has not been tested whether it was intended or not. See https://github.com/terraform-linters/tflint/pull/1218
- Untested binaries
  - For example, BSD-based OS is not tested

To solve this problem, we would like to end support for hard-to-test or infrequently used binaries and clearly determine which os/arch to support.

One piece of information that can be helpful is the types of binaries that Terraform supports.
https://www.terraform.io/downloads.html

- macOS
  - amd64
  - arm64
- FreeBSD
  - 386
  - amd64
  - arm
- Linux
  - 386
  - amd64
  - arm
  - arm64
- OpenBSD
  - 386
  - amd64
- Solaris
  - amd64
- Windows
  - 386
  - amd64

Also, the number of downloads is helpful information. Below is the number of downloads in the last 30 releases:

```
tflint_darwin_arm64.zip: 57
tflint_freebsd_386.zip: 304
tflint_linux_arm64.zip: 101
tflint_netbsd_arm.zip: 305
tflint_windows_amd64.zip: 20940
tflint_windows_arm64.zip: 0
tflint_freebsd_arm64.zip: 10
tflint_openbsd_amd64.zip: 310
tflint_darwin_386.zip: 225
checksums.txt: 21618
tflint_darwin_amd64.zip: 12661
tflint_freebsd_amd64.zip: 321
tflint_windows_386.zip: 859
tflint_openbsd_386.zip: 303
checksums.txt.sig: 13596
tflint_freebsd_arm.zip: 305
tflint_linux_386.zip: 12147
tflint_linux_amd64.zip: 1252744
tflint_linux_arm.zip: 459
tflint_netbsd_386.zip: 308
tflint_netbsd_amd64.zip: 310
tflint_openbsd_arm.zip: 297
tflint_openbsd_arm64.zip: 8
tflint_windows_arm.zip: 13
```

Script:

<details>

```go
package main

import (
	"context"
	"fmt"

	"github.com/google/go-github/v39/github"
)

func main() {
	client := github.NewClient(nil)
	releases, _, err := client.Repositories.ListReleases(context.Background(), "terraform-linters", "tflint", nil)
	if err != nil {
		panic(err)
	}

	results := map[string]int{}
	for _, release := range releases {
		for _, asset := range release.Assets {
			results[*asset.Name] = results[*asset.Name] + *asset.DownloadCount
		}
	}

	for name, count := range results {
		fmt.Printf("%s: %d\n", name, count)
	}
}
```

</details>

Based on these results, we would like to support only the following binaries:

- macOS
  - amd64
  - arm64
- Linux
  - 386
  - amd64
  - arm
  - arm64
- Windows
  - 386
  - amd64
